UI/window-basic-main.cpp

```diff
@@ -467,6 +469,25 @@ static obs_data_t *GenerateSaveData(obs_data_array_t *sceneOrder,
 	return saveData;
 }
 
+void OBSBasic::DoRestart(bool restart)
+{
+	doRestart = restart;
+}
+
+void OBSBasic::Restart()
+{
+	QMessageBox::StandardButton button =
+				OBSMessageBox::question(this,
+						QTStr("Restart"),
+						QTStr("ProgramRestarted"));
+
+	if (button == QMessageBox::Yes) {
+		qApp->quit();
+		QProcess::startDetached(qApp->arguments()[0],
```

Comment:
This seems to open up a race condition where the new process tries to start before the old one is completely done shutting down. Perhaps a new startup flag is needed to make OBS wait for any other OBS processes to exit before trying to load configs etc.